### PR TITLE
[TINKERPOP-2838] Adds Tests for User Agent in GLVs

### PR DESCRIPTION
**Pull request by Cole-Greer**
https://issues.apache.org/jira/browse/TINKERPOP-2838

This is my final PR regarding the User Agent for web socket connections. It simply adds a couple of simple tests in each GLV validating that the user agent is behaving correctly.

### CODE REVIEW

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Driver/GremlinClientBehaviorIntegrationTests.cs [hunk 1]
```diff
@@ -84,5 +84,37 @@ await gremlinClient.SubmitWithSingleResultAsync<Vertex>(RequestMessage.Build("1"
...
         }
+
+        [Fact]
+        public async Task ShouldIncludeUserAgentInHandshakeRequest()
+        {
+            var sessionId = Guid.NewGuid().ToString();
+            var poolSettings = new ConnectionPoolSettings {PoolSize = 1};
+
+            var gremlinServer = new GremlinServer(TestHost, Settings.Port);
+            var gremlinClient = new GremlinClient(gremlinServer, messageSerializer: Serializer,
```
[the remark this answers is not in the document]
Comment (Cole-Greer) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Driver/GremlinClientBehaviorIntegrationTests.cs]: Sure, I will add this

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Driver/GremlinClientBehaviorIntegrationTests.cs [hunk 2]
```diff
@@ -84,5 +84,37 @@ await gremlinClient.SubmitWithSingleResultAsync<Vertex>(RequestMessage.Build("1"
             Assert.NotNull(response2);
             Assert.Equal(1, gremlinClient.NrConnections);
         }
+
+        [Fact]
+        public async Task ShouldIncludeUserAgentInHandshakeRequest()
+        {
+            var sessionId = Guid.NewGuid().ToString();
+            var poolSettings = new ConnectionPoolSettings {PoolSize = 1};
```
[the remark this answers is not in the document]
Comment (Cole-Greer) [reply gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Driver/GremlinClientBehaviorIntegrationTests.cs]: Thanks, I have removed unnecessary settings

File: gremlin-dotnet/src/Gremlin.Net/Process/Utils.cs [hunk 3]
```diff
@@ -32,8 +32,11 @@ namespace Gremlin.Net.Process
     /// <summary>
     /// Contains useful methods that can be reused across the project. 
     /// </summary>
-    internal static class Utils
+    public static class Utils
```
[the remark this answers is not in the document]
Comment (Cole-Greer) [reply gremlin-dotnet/src/Gremlin.Net/Process/Utils.cs]: Thanks for suggestions Florian. I have implemented this one using the same key file as is being used for `Gremlin.Net.UnitTest`. Please let me know if I should instead be generating new keys for this.

[the remark this answers is not in the document]
Comment (FlorianHockmann) [reply gremlin-dotnet/src/Gremlin.Net/Process/Utils.cs]: Using the same key file definitely makes sense. We're also already using the same file for the [source project itself](https://github.com/apache/tinkerpop/blob/master/gremlin-dotnet/src/Gremlin.Net/Gremlin.Net.csproj#L60).

**Review by FlorianHockmann — APPROVED** [review]
VOTE +1

**Comment by vkagamlyk** [comment]
VOTE +1

**Comment by xiazcy** [comment]
VOTE +1

**Pull request merged**